File: src/main/java/com/jloom/util/ProjectPaths.java
package com.jloom.util;

import java.nio.file.Files;
import java.nio.file.Path;

public final class ProjectPaths {

    private ProjectPaths() { }

    public static boolean isEmpty(Path dir) {
        if (!Files.isDirectory(dir)) return true;
        try (var stream = Files.list(dir)) {
            return stream.findAny().isEmpty();
        } catch (Exception _) {
            return false;
        }
    }

    public static void requireEmpty(Path target) {
        if (!isEmpty(target)) {
            throw new IllegalArgumentException(
                    "'" + target.toAbsolutePath() + "' already exists and isn't empty — "
                            + "pass a different --name or remove it first.");
        }
    }
}

File: bun/src/util/projectPaths.ts
import { existsSync, readdirSync, statSync } from "node:fs";
import path from "node:path";

export function isEmpty(dir: string): boolean {
  if (!existsSync(dir) || !statSync(dir).isDirectory()) {
    return true;
  }
  try {
    return readdirSync(dir).length === 0;
  } catch {
    return false;
  }
}

export function requireEmpty(target: string): void {
  if (!isEmpty(target)) {
    throw new Error(
      `'${path.resolve(target)}' already exists and isn't empty — pass a different --name or remove it first.`,
    );
  }
}
